javalib/src/main/scala/java/net/HttpCookie.scala

```diff
@@ -0,0 +1,374 @@
+package java.net
+
+import java.util.ArrayList
+
+object HttpCookie {
+
+  def domainMatches(domain: String, host: String): Boolean = {
+    val effectiveHost = if (host.indexOf('.') == -1) {
+      host + ".local"
+    } else
+      host
+
+    val domainL = domain.toLowerCase
+    val hostL = host.toLowerCase
+
+    val exactMatch = domainL == hostL
+    // a host is a subdomain if it ends with the domain, and
+    // the character preceeding the domain suffix is a dot
+    val hostIsSubdomain = hostL.endsWith("." + domainL)
+
+    domainL == hostL || hostIsSubdomain
+
+  }
+
+  private[HttpCookie] val Discard = "Discard"
```

Comment:
Please pardon my lack of understanding.
What does specifying the scope `[HttpCookie]` add here?
How does it differ from `private val`.

I thought that companion classes and corresponding object
could reference each other's private variables.  Yes, the 
class must either import the symbols of the companion Object
or do someething like `HttpCookie.Discard`.

I just checked and one can have `final` vals of type `String`.  I know
that adding `final` helps the SN compiler treat numeric vals as a
compile time constant.  I do not know if that works with String.
Perhaps someone with better knowledge of Scala Native internals
can chime in.

Anyway I think `final` will not hurt.